chevah/keycert/ssh.py

```diff
@@ -299,7 +299,7 @@ def fromString(cls, data, type=None, passphrase=None):
             # Most probably some parts are missing from the key, so
             # we consider it too short.
             raise BadKeyError('Key is too short.')
-        except (struct.error,  binascii.Error, TypeError):
+        except (struct.error,  binascii.Error, TypeError, AssertionError):
```

Comment:
It is ok to raise a non-BadKeyError, just don't use the AssertionError for this :)

we are using AssertionError for 'imposibile' error conditions and for preventing using an API in in the wrong way.

we don't use AssertionError for input validation.